cpp/oneapi/dal/detail/common.hpp

```diff
@@ -72,4 +72,76 @@ Object make_from_pointer(typename Object::pimpl::element_type* pointer) {
     return pimpl_accessor().template make_from_pointer<Object>(pointer);
 }
 
+constexpr std::int64_t get_data_type_size(data_type t) {
+    if (t == data_type::float32) {
+        return sizeof(float);
```

Comment:
Shouldn't this `if`s be marked as `constexpr`? at least for consistency with `make_data_type_impl` function?

Reply:
It cannot be marked as `constexpr` since the value of `t` is determined at runtime. But this function can be evaluated at compile time if `t` is known at compile time, so the function itself marked as `constexpr`.